pkg/controller/nodelifecycle/scheduler/taint_manager.go

```diff
@@ -58,6 +60,32 @@ type podUpdateItem struct {
 	newTolerations []v1.Toleration
 }
 
+func (n *nodeUpdateItem) name() string {
+	if n.newNode != nil {
+		return n.newNode.ObjectMeta.Name
+	}
+	if n.oldNode != nil {
+		return n.oldNode.ObjectMeta.Name
+	}
+	return ""
+}
+
+func (p *podUpdateItem) nodeName() string {
```

Comment:
Should we use `namespace/name` for pod? The nodename maybe changed from empty to some-host by bind.

Reply:
We want the pods to be processed by the same worker as their node is processed (to ensure that node updates are processed first).

Note that pods that are not assigned yet are ignored later anyway:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/nodelifecycle/scheduler/taint_manager.go#L346

So that is fine.

Reply:
OK to me :)